wfsim/strax_interface.py

```diff
@@ -100,31 +109,39 @@ def read_optical(c):
     if not c['neutron_veto']:
         ins = ins[ins["amp"] > 0]
 
-    return ins, channels, timings
-
+    #Slightly weird here. Entry_stop is not in the regular config, so if it's not skip this part
+    if c.get('event_stop',-1)!=-1:
+        mask = (ins['g4id']<c['event_stop'])&(ins['g4id']>=c['event_start'])
 
-def read_optical_nveto_with_cut(c, e):
+    return ins[mask], channels[mask], timings[mask]
+        
+def _read_optical_nveto(config, events):
+    '''Helper function for nveto to read photon channels and timings from G4 and apply QE's
+    :params c: dict, wfsim configuration
+    :params e: g4 root file
+    
+    returns 2 nested arrays
+    '''
     nV_pmt_id_offset = 2000
-    channels = [[channel - nV_pmt_id_offset for channel in array] for array in e["pmthitID"].array(library="np")]
-    timings = e["pmthitTime"].array(library="np")*1e9
     constant_hc = 1239.841984 # (eV*nm) to calculate (wavelength lambda) = h * c / energy
-    wavelengths = [[constant_hc / energy for energy in array] for array in e["pmthitEnergy"].array(library="np")]
-
-    collection_efficiency = c['nv_pmt_ce_factor']
-    resource = load_config(c)
+    channels = [[channel - nV_pmt_id_offset for channel in array] for array in events["pmthitID"].array(library="np")]
+    timings = events["pmthitTime"].array(library="np")*1e9
+    wavelengths = [[constant_hc / energy for energy in array] for array in events["pmthitEnergy"].array(library="np")]
+    collection_efficiency = config['nv_pmt_ce_factor']
```

Comment:
Out of curiosity, what kind od collection efficiency is this? I though QE would be enough.

Reply:
CE at the 1st dynode, which comes after (and isn't included) in the QE.

Reply:
Interesting, how do you measure/estimate this without automatically including it in the QE?

Reply:
We need a Japanese for this 😉 
cc @mzks

You can find some good references for the (TPC) PMTs characterization in this note: https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:mc:g1_nt_projection

Reply:
Okay I see it is a relative value.  Thanls

Reply:
Please let me explain a bit more about this part.
Tha PMT acceptance is generally considered as Q.E. (lambda, theta, phi) * C.E.(theta, phi) in each PMT. The only way to have them is just measurement, but we didn't. They are measured as a coupled value. I guess that Q.E. is not enough, but we don't know the entire value even if we considered optical behavior on our MC. I would like to keep tuning parameter at here as 'ce' for our future. However, it is super-fine tuning thus I afraid to make some bias at this part. I would like to use the value as 1.0 now, but we need to discuss some days.